### PR TITLE
Linear algebraic systems

### PR DESCRIPTION
**Pull request by mforets**
This PR adds yet another class from [HybridSystems.jl](https://github.com/blegat/HybridSystems.jl/blob/master/src/systems.jl#L71), the `LinearAlgebraic` ones, for the continuous and discrete cases.

### CODE REVIEW

File: src/continuous.jl [hunk 1]
```diff
@@ -90,3 +90,52 @@ end
...
+- `A` -- square matrix
+- `E` -- square matrix
+"""
+struct LinearAlgebraicContinuousSystem{T, MT <: AbstractMatrix{T}} <: AbstractContinuousSystem
+    A::MT
+    E::MT
+end
+function LinearAlgebraicContinuousSystem{T, MT <: AbstractMatrix{T}}(A::MT, E::MT)
+    @assert Base.LinAlg.checksquare(A) == Base.LinAlg.checksquare(E)
+    return LinearAlgebraicContinuousSystem{T, MT}(A, E)
```
Comment (schillic) [inline src/continuous.jl]: Maybe define [such constructors inside the struct](https://docs.julialang.org/en/stable/manual/constructors/#Inner-Constructor-Methods-1).

File: src/continuous.jl [hunk 2]
```diff
@@ -90,3 +90,52 @@ end
...
+end
+statedim(s::LinearAlgebraicContinuousSystem) = Base.LinAlg.checksquare(s.A)
```
Comment (schillic) [inline src/continuous.jl]: No need to call the more expensive `Base.LinAlg.checksquare(s.A)` here: The constructor ensures that the matrix is square.

File: src/continuous.jl [hunk 3]
```diff
@@ -90,3 +90,52 @@ end
...
+inputdim(s::LinearAlgebraicContinuousSystem) = 0
```
Comment (schillic) [inline src/continuous.jl]: Minor: You can omit the name `s` here.

File: src/continuous.jl (hunk 1 above)
Comment (mforets) [reply src/continuous.jl]: if it's outside, this is a bit more flexible in the sense that the user can provide more outer constructors.. just a thought..

File: src/continuous.jl (hunk 2 above)
Comment (mforets) [reply src/continuous.jl]: done (the `statedim` changes for older types are better a separate PR)

File: src/continuous.jl [hunk 4]
```diff
@@ -111,7 +111,7 @@ function LinearAlgebraicContinuousSystem{T, MT <: AbstractMatrix{T}}(A::MT, E::M
     @assert Base.LinAlg.checksquare(A) == Base.LinAlg.checksquare(E)
     return LinearAlgebraicContinuousSystem{T, MT}(A, E)
 end
-statedim(s::LinearAlgebraicContinuousSystem) = Base.LinAlg.checksquare(s.A)
+statedim(s::LinearAlgebraicContinuousSystem) = size(A, 1)
```
Comment (schillic) [inline src/continuous.jl]: `A` -> `s.A` (also below)

File: src/continuous.jl [hunk 5]
```diff
@@ -90,3 +90,52 @@ end
...
+
+Continuous-time linear algebraic system of the form
+```math
+E x' = A x.
+```
+
+### Fields
+
+- `A` -- square matrix
+- `E` -- square matrix
```
Comment (blegat) [inline src/continuous.jl]: The matrices do not need to be square. But they should have the same dimension though.

File: src/continuous.jl (hunk 1 above)
Comment (schillic) [reply src/continuous.jl]: For me the idea is to perform this check *always*. Breaking type assumptions is usually not "flexible" 🤕 A user can still provide more outer constructors, they just have to call the inner one.

File: src/continuous.jl (hunk 5 above)
Comment (mforets) [reply src/continuous.jl]: you're right,  i'll use `size()` for the check

**Pull request merged**